rabbus.go

```diff
@@ -224,9 +226,12 @@ func (r *rabbus) register() {
 }
 
 func (r *rabbus) produce(m Message) {
-	if err := r.ch.ExchangeDeclare(m.Exchange, m.Kind, r.config.Durable, false, false, false, nil); err != nil {
-		r.emitErr <- err
-		return
+	if _, ok := r.exDeclared[m.Exchange]; !ok {
```

Comment:
**sorry for the delay I'm without internet connection at home ATM ** Indeed, the produce method is safe behind the go channels